### ISSUE TITLE
At checked "Pending review", generate unintentional revision

### ISSUE THREAD

**Issue by akiya64** [WordPress/gutenberg#22259]
At checked "Pending review", generate unintentional revision by showing Preview.

## To reproduce
1. Create post with Custom Fields.
2. Check "Pending review".
3. Edit contents or custom fields.
3. Show preview by click "Preview" button.
4. Move dashboard and editing same post again, or reload page after click "Preview".
4. Generated new "auto save", inclements revision but I did not change contents.

## Expected behavior
I want not generate new revision, if contents has no change.
On WP5.4.0, generated new "auto save" by showing preview avoid all custom fields.
So I hope that keep custom fields at checked "pending review".
On WP 5.4.1 it seems keep custom fields in generated revision by showing Preview.

## Screenshots
At reproduce step 6.
![2020-05-11_17h03_25](https://user-images.githubusercontent.com/14228487/81543032-81d57a80-93b0-11ea-9d94-4d17aff70d70.png)

![2020-05-11_17h37_39](https://user-images.githubusercontent.com/14228487/81543049-8732c500-93b0-11ea-89b6-0b9a6f0aa589.png)

## Editor version
- WordPress version: 5.4.1
- default Gutenberg

## Desktop
 - OS: Windows 10 Pro 1903
 - Browser FireFox
 - Version 76.0.1

**Comment by peterwilsoncc** [comment]
This was able to be reproduced in a triage session today.

As in the screen shot above, it's reproducible without a custom fields plugin/custom meta boxes and occurs with WP Core's custom fields. 

@talldan mentioned it may be because the custom fields form prevents Gutenberg from knowing if the post is dirty so generates an autosave whenever the preview button is pressed.

**Comment by akiya64** [comment]
I was re-attacked this issue in WordCamp Japan 2021 contributer day.
I tested in WordPress 5.7.2 and Gutenberg latest trunk again, **this problem is not reploduce**.

I found PR, [Fix unexpected autosave for published posts by youknowriad · Pull Request #27942](https://github.com/WordPress/gutenberg/pull/27942) in Gutenberg 9.8.0 release note.
Maybe this issue was fixed by that PR.

Related issue #27939
